src/abTestingInfra/ABTestingContext/index.tsx

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import { AB_TEST_PAYLOAD_PREFIX } from '../utils';
+import { useRouter } from 'next/router';
+import { ABTestingPayload } from '../types';
+
+const defaultABTestingContext: ABTestingPayload = {};
+const ABTestingContext = React.createContext(defaultABTestingContext);
```

Comment:
Can we move this lower in the file, closer to the `Provider` definition & subsequent exports?

Also, what about using `const { Provider, Consumer } = ...`?

Reply:
Done!